Documentation/In-depth/ComposerJson/Index.rst

```diff
@@ -0,0 +1,288 @@
+.. include:: /Includes.txt
+
+
+.. _project_composer.json:
+
+=============
+composer.json
+=============
+
+In Composer, everything is a package, but there are differences between
+a "project" and a "library" and the best practices that are used within
+the file :file:`composer.json` for a project or library.
+
+This page explains the basics for your **project** (TYPO3 installation)
+:file:`composer.json`, while :file:`composer.json` for libraries (or more
+specifically for TYPO3 extensions) is handled in TYPO3 Explained
+:ref:`t3coreapi:composer-json`
+
+In a Composer based TYPO3 installation, we typically have a file structure
+like this:
+
+
+.. code-block:: none
+
+   project-root/
+   ├── composer.json
+   ├── composer.lock
+   └── public
+       ├── fileadmin
+       ├── typo3
+       ├── typo3conf
+       └── ...
+
+You may have used the `typo3/cms-base-distribution` package (as explained in
+:ref:`install-via-composer`) which created a :file:`composer.json`
+for you or you may want to create a :file:`composer.json` for your project from
+scratch.
+
+.. _project_composer.json-example:
+
+Example composer.json
+=====================
+
+.. code-block:: json
+
+   {
+      "name": "vendor/mysite",
+      "description": "",
+      "license": ["GPL-2.0-or-later"],
+      "type": "project",
+      "authors": [
+         {
+            "name": "Author name",
+            "email": "nouser@example.com"
+         }
+      ],
+      "require": {
+         "helhum/typo3-console": "^6.4.0",
+         "typo3/cms-backend": "^11.0.0",
+         "typo3/cms-belog": "^11.0.0",
+         "typo3/cms-beuser": "^11.0.0",
+         "typo3/cms-core": "^11.0.0",
+         "typo3/cms-dashboard": "^11.0.0",
+         "typo3/cms-extbase": "^11.0.0",
+         "typo3/cms-extensionmanager": "^11.0.0",
+         "typo3/cms-felogin": "^11.0.0",
+         "typo3/cms-filelist": "^11.0.0",
+         "typo3/cms-fluid": "^11.0.0",
+         "typo3/cms-fluid-styled-content": "^11.0.0",
+         "typo3/cms-form": "^11.0.0",
+         "typo3/cms-frontend": "^11.0.0",
+         "typo3/cms-impexp": "^11.0.0",
+         "typo3/cms-info": "^11.0.0",
+         "typo3/cms-install": "^11.0.0",
+         "typo3/cms-recordlist": "^11.0.0",
+         "typo3/cms-rte-ckeditor": "^11.0.0",
+         "typo3/cms-seo": "^11.0.0",
+         "typo3/cms-setup": "^11.0.0",
+         "typo3/cms-sys-note": "^11.0.0",
+         "typo3/cms-t3editor": "^11.0.0",
+         "typo3/cms-tstemplate": "^11.0.0",
+         "typo3/cms-viewpage": "^11.0.0"
+      },
+      "autoload": {
+         "psr-4": {},
+         "classmap": []
+      },
+      "config": {
+         "platform": {
+            "php": "7.4.1"
```

Comment:
Personally I see this as dangerous if not added intentionally. People will copy this, set up everything for PHP 7.4 only to notice afterwards that their remote server has PHP 7.2/8.0. Instead `php` should be a reqular requirement, see above.

Reply:
Ok, I removed it in composer.json and use not the version but a placeholder text in the config snippet below.